### PR TITLE
Refactor: create new forms using block models for translatable strings

### PR DESCRIPTION
**Pull request by jonwaldstein**
Resolves: #7035

### The problem
Previously we have been generating new forms using our static `blocks.json` file.  This was problematic with translation because we are simply storing english words in the database.  WordPress translation only works with static strings, so once you store your strings in the database - they will remain untranslated until translated manually.

### The solution
This updates the generation of a new form to use block models instead of the static `blocks.json` file.  This is necessary to include translatable strings during the initial form creation as all user-facing strings are wrapped in translation functions.  

Of course this isn't perfect, the strings would need to be pre-translated before creating a new form so the translated words are stored in the database.  Yet, this is a good start as at least the strings will be included in the translation file.

## Affects

- New v3 form creation

## Visuals

**Example of pre-translated string in italiano and creating a new form:**

Screenshot 2023-10-19 at 3 32 29 PM

Screenshot 2023-10-19 at 3 33 28 PM

## Testing Instructions

QA build: 
https://github.com/impress-org/givewp/actions/runs/6617533934

- First make sure you can create a new form without any issues
- Then try creating a form in a different language (with default section strings translated)
for example: `How much would you like to donate today?`
- Those default section strings should not be in english.

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [ ] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [ ] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205761241674736

### CODE REVIEW

File: src/FormBuilder/Routes/CreateFormRoute.php [hunk 1]
```diff
@@ -30,18 +30,14 @@ public function __invoke()
...
-
                 $form = new DonationForm([
                     'title' => __('GiveWP Donation Form', 'give'),
                     'status' => DonationFormStatus::DRAFT(),
                     'settings' => FormSettings::fromArray([
                         'enableDonationGoal' => true,
                         'goalAmount' => 1000,
                     ]),
-                    'blocks' => BlockCollection::fromJson($blocksJson)
+                    'blocks' => (new GenerateDefaultDonationFormBlockCollection())(),
```
Comment (glaubersilva) [inline src/FormBuilder/Routes/CreateFormRoute.php:40]: We need to add an unreleased tag about these changes.

File: src/Framework/Blocks/BlockModel.php [hunk 2]
```diff
@@ -101,17 +121,12 @@ public function getShortName(): string
...
-                $blockData['innerBlocks'])
-        ) : new BlockCollection([]);
-
         return new BlockModel(
             $blockData['name'],
             !empty($blockData['clientId']) ? $blockData['clientId'] : wp_generate_uuid4(),
             !empty($blockData['isValid']) ? $blockData['isValid'] : true,
             !empty($blockData['attributes']) ? $blockData['attributes'] : [],
-            $innerBlocks
+            $blockData['innerBlocks'] ?? []
```
Comment (glaubersilva) [inline src/Framework/Blocks/BlockModel.php:129]: Missing an unreleased tag here as well.

**Review by glaubersilva — CHANGES_REQUESTED** [review]
@jonwaldstein Great work! I left a couple of comments related to unreleased tags. 

As a note, in my tests, I realized that a few strings weren't translated - check the attached screenshots for reference. However, it happened because the Italian translation file ( `give-it_IT.po` ) didn't have a few strings present in the source file ( `give.pot` ) like, for example, the `Who's Giving Today?` (with capitalized letters), so after using the Loco sync feature and translating this string, the new forms also included this translation. 

screencapture-givewp-local-donations-modulo-di-donazione-di-givewp-2023-10-23-11_13_21

image

screencapture-givewp-local-donations-modulo-di-donazione-di-givewp-2023-10-23-12_43_11

**Comment by jonwaldstein** [comment]
@glaubersilva yes what you are experiencing is expected, the strings will still need to be present in the translation file and translated before creating a new form.

>Of course this isn't perfect, the strings would need to be pre-translated before creating a new form so the translated words are stored in the database. Yet, this is a good start as at least the strings will be included in the translation file.

File: src/FormBuilder/Routes/CreateFormRoute.php (hunk 1 above)
Comment (jonwaldstein) [reply src/FormBuilder/Routes/CreateFormRoute.php:41]: resolved a2ce5081

**Comment by jonwaldstein** [comment]
@glaubersilva feedback addressed, ready for re-review.

**Review by glaubersilva — APPROVED** [review]
@jonwaldstein Thanks for the updates, looks good to me!

**Review by rickalday — APPROVED** [review]
Passed manual QA tests

**Pull request merged**